Replicator/tests/ReplicatorLoopbackTest.cc

```diff
@@ -2060,3 +2064,127 @@ TEST_CASE_METHOD(ReplicatorLoopbackTest, "Conflict Includes Rev", "[Push][Sync]"
+    }
+#    endif
+    SECTION("Server Pull") {
+        testMode = TestMode::ServerPull;
+        createRev(_collDB1, "doc"_sl, kRevID, kFleeceBody);
+        clientOpts = std::make_shared<Replicator::Options>(Replicator::Options::passive(_collSpec));
+        serverOpts = std::make_shared<Replicator::Options>(Replicator::Options::pulling(kC4OneShot, _collSpec));
+        serverOpts->setDelayChangesResponse(true);
+        _expectedDocumentCount = 1;
+    }
+#    ifdef ACTIVE_PUSH_REPLACEMENTREVS
+    SECTION("ClientPush Do not send Replacement Rev") {
+        testMode = TestMode::ClientPush;
+        createRev(_collDB1, "doc"_sl, kRevID, kFleeceBody);
+        clientOpts = std::make_shared<Replicator::Options>(Replicator::Options::pushing(kC4OneShot, _collSpec));
+        serverOpts = std::make_shared<Replicator::Options>(Replicator::Options::passive(_collSpec));
+        serverOpts->setDelayChangesResponse(true);
+        _expectedDocumentCount = 0;
+        // Disable the Puller from requesting replacementRev
+        serverOpts->setDisableReplacementRevs(true);
+    }
+#    endif
+    SECTION("ServerPull Do not send Replacement Rev") {
+        testMode = TestMode::ServerPull;
+        createRev(_collDB1, "doc"_sl, kRevID, kFleeceBody);
+        clientOpts = std::make_shared<Replicator::Options>(Replicator::Options::passive(_collSpec));
+        serverOpts = std::make_shared<Replicator::Options>(Replicator::Options::pulling(kC4OneShot, _collSpec));
+        serverOpts->setDelayChangesResponse(true);
+        _expectedDocumentCount = 0;
+        serverOpts->setDisableReplacementRevs(true);
+    }
+
+    string testModeStr;
+    switch ( testMode ) {
+        case TestMode::ClientPush:
+            testModeStr = "ClientPush";
+            break;
+        case TestMode::ClientPull:
+            testModeStr = "ClientPull";
+            break;
+        case TestMode::ServerPush:
+            testModeStr = "ServerPush";
+            break;
+        case TestMode::ServerPull:
+            testModeStr = "ServerPull";
+            break;
+    }
+
+    REQUIRE(runReplicatorsAsync(*clientOpts, *serverOpts));
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    switch ( testMode ) {
+        case TestMode::ClientPush:
+            mutateDoc(_collDB1, "doc"_sl, [](MutableDict props) { props["birthday"_sl] = "1964-11-28"_sl; });
+            break;
+        case TestMode::ClientPull:
+            mutateDoc(_collDB2, "doc"_sl, [](MutableDict props) { props["birthday"_sl] = "1964-11-28"_sl; });
+            break;
+        case TestMode::ServerPush:
+            mutateDoc(_collDB2, "doc"_sl, [](MutableDict props) { props["birthday"_sl] = "1964-11-28"_sl; });
+            break;
+        case TestMode::ServerPull:
+            mutateDoc(_collDB1, "doc"_sl, [](MutableDict props) { props["birthday"_sl] = "1964-11-28"_sl; });
+            break;
+    }
+    waitForReplicators(*clientOpts, *serverOpts);
+    switch ( testMode ) {
```

Comment:
I would also check the current rev. For example, w.r.t. client pull, remote seq = 1, but current rev = 2-x. Still better, do a new pull, and verify that puller receive 2-x, because remote seq is 1, and puller already has 2-x, and replicator is done without actual BLIP rev msg.

Reply:
Thanks, I've done so in the latest commit